hack/kcp-edge.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bash
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: $0 (start or stop | --user dev or kit | -v
+
+# Purpose: deploy the kcp-edge platform. The following components are created"
+#           (a) 1 kcp workspace: edge service provider workspace (espw)
+#           (b) 3 kcp-edge controllers: edge-scheduler, mailbox-controller and placement-translator
+
+# Assumption: kcp server is running.
+
+# This script requires the edge-mc binaries mailbox-controller, scheduler, placement-translator
+# already exist at edge-mc/bin/
+
+# This script requires that the parent directory is edge-mc to run in dev mode
+
+
+set -e
+
+install=""
+verbosity=0
+espw_name="espw"
+user_type="dev"
+
+while (( $# > 0 )); do
+    if [ "$1" == "start" ]; then
+        install=1
+    elif [ "$1" == "stop" ]; then
+        install=0
+    elif [ "$1" == "-v" ]; then
+        verbosity=1
+    elif [ "$1" == "--user" ]; then
+        user_type=$2
```

Comment:
Need a `shift` here.

How about checking the validity of the arguments before making any changes?

Reply:
Fixed